cassandra/cluster.py

```diff
@@ -372,6 +373,14 @@ def auth_provider(self, value):
     If set to :const:`None`, there will be no timeout for these queries.
     """
 
+    idle_heartbeat_interval = 30
```

Comment:
I don't think there's a lot of science behind it, but the general notion that it should be sufficiently high that there is little to no chance of false negatives (this value taken to align with other drivers' default settings). We don't want a bunch of connection thrashing 'by default'. 
Many modes of connection failure would detect the problem on send, which would be much more responsive. 
WRT missing events, we can at least rely on the new connection to refresh topology when it comes up.
Of course users are free to tighten this up if it suits their environment.